<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Advanced Online Store Simulation</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true">
     QWidget {
      background-color:rgb(221, 221, 221);
      border-radius: 10px;
      padding: 20px;
     }
    </string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
   
    <!-- Search Bar -->
    <item>
     <widget class="QLineEdit" name="searchBar">
      <property name="placeholderText">
       <string>Note:</string>
      </property>
      <property name="styleSheet">
       <string notr="true">
        QLineEdit {
         background-color:rgb(170, 145, 145);
         border: 1px solid #DDD;
         padding: 10px;
         border-radius: 5px;
         font-size: 14px;
        }
        QLineEdit:focus {
         border: 1px solidrgb(168, 245, 242);
         box-shadow: 0 0 5px rgba(103, 123, 122, 0.6);
        }
       </string>
      </property>
     </widget>
    </item>

    <!-- Product Table -->
    <item>
     <widget class="QTableWidget" name="productTable">
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <property name="columnCount">
       <number>5</number>
      </property>
      <property name="horizontalHeaderLabels" stdset="0">
       <stringlist>
        <string>Name</string>
        <string>Price</string>
        <string>Stock</string>
        <string>Description</string>
        <string>Quantity</string>
       </stringlist>
      </property>
      <property name="styleSheet">
       <string notr="true">
        QTableWidget {
         background-color: #FFFFFF;
         color: #333333;
         border: 1px solid #DDD;
         font-size: 14px;
         border-radius: 5px;
        }
        QTableWidget::item {
         padding: 8px;
         border-bottom: 1px solid #EEE;
        }
        QTableWidget::item:selected {
         background-color: #D9E6F3;
         color: #1D60A3;
        }
        QTableWidget::horizontalHeader {
         background-color: #F8F8F8;
         color: #555;
         padding: 8px;
         border-bottom: 1px solid #DDD;
        }
       </string>
      </property>
     </widget>
    </item>

    <!-- Customer Buttons -->
    <item>
     <widget class="QFrame" name="customerButtonsFrame">
      <property name="styleSheet">
       <string notr="true">
        QFrame {
         background-color: #EAF4F4;
         border-radius: 10px;
         padding: 15px;
         margin-bottom: 20px;
        }
        QPushButton {
         background: #5C8D8B;
         color: white;
         border-radius: 10px;
         padding: 14px 22px;
         font-size: 16px;
         border: none;
         transition: all 0.3s ease;
         box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
        }
        QPushButton:hover {
         background: #4A7672;
         transform: scale(1.05);
         box-shadow: 0 6px 12px rgba(0, 0, 0, 0.1);
        }
        QPushButton:pressed {
         background: #3B615E;
         transform: scale(0.98);
         box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
        }
       </string>
      </property>
      <layout class="QVBoxLayout" name="customerButtonsLayout">
       <item>
        <widget class="QPushButton" name="addToCartButton">
         <property name="text">
          <string>Add to Cart</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="viewCartButton">
         <property name="text">
          <string>View Cart</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="removeProductFromCartButton">
         <property name="text">
          <string>Remove from Cart</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="applyDiscountButton">
         <property name="text">
          <string>Apply Discount</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="applyPromoCodeButton">
         <property name="text">
          <string>Apply PromoCode</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="finalizeOrderButton">
         <property name="text">
          <string>Finalize Order</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

    <!-- Seller Buttons -->
    <item>
     <widget class="QFrame" name="sellerButtonsFrame">
      <property name="styleSheet">
       <string notr="true">
        QFrame {
         background-color:rgb(255, 239, 190);
         border-radius: 10px;
         padding: 15px;
         margin-bottom: 20px;
        }
        QPushButton {
         background:rgb(254, 190, 88);
         color: white;
         border-radius: 10px;
         padding: 14px 22px;
         font-size: 16px;
         border: none;
         transition: all 0.3s ease;
         box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
        }
        QPushButton:hover {
         background: #E67E22;
         transform: scale(1.05);
         box-shadow: 0 6px 12px rgba(0, 0, 0, 0.1);
        }
        QPushButton:pressed {
         background: #D35400;
         transform: scale(0.98);
         box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
        }
       </string>
      </property>
      <layout class="QVBoxLayout" name="sellerButtonsLayout">
       <item>
        <widget class="QPushButton" name="addProductButton">
         <property name="text">
          <string>Add Product</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="editProductButton">
         <property name="text">
          <string>Edit Product</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="removeProductButton">
         <property name="text">
          <string>Remove Product</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="viewOrdersButton">
         <property name="text">
          <string>View Orders</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

    <!-- Footer -->
    <item>
     <widget class="QFrame" name="footerFrame">
      <property name="styleSheet">
       <string notr="true">
        QFrame {
         background-color: #F0F0F0;
         border-radius: 10px;
         padding: 10px;
         text-align: center;
        }
        QLabel {
         font-size: 12px;
         color: #777;
        }
       </string>
      </property>
      <layout class="QHBoxLayout" name="footerLayout">
       <item>
        <widget class="QLabel" name="footerLabel">
         <property name="text">
          <string>Advanced Online Store Simulation</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
